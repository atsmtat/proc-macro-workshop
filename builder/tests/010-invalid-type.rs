// Ensure that macro reports a reasonable error message when the
// caller use it for a type which is not supported.

use derive_builder::Builder;

#[derive(Builder)]
pub struct Id(u32);

#[derive(Builder)]
pub struct Unit();

#[derive(Builder)]
pub enum Command {
    Left(u32),
    Right(u32),
}

fn main() {}
