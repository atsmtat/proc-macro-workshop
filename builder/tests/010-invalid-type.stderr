error: Struct with Builder macro must use named fields
 --> tests/010-invalid-type.rs:7:14
  |
7 | pub struct Id(u32);
  |              ^^^^^

error: Struct with Builder macro must use named fields
  --> tests/010-invalid-type.rs:10:16
   |
10 | pub struct Unit();
   |                ^^

error: Builder macro is allowed only on a struct type
  --> tests/010-invalid-type.rs:12:10
   |
12 | #[derive(Builder)]
   |          ^^^^^^^
   |
   = note: this error originates in the derive macro `Builder` (in Nightly builds, run with -Z macro-backtrace for more info)
